### ISSUE TITLE
Torch train-prediction pipeline causes excessive memory usage

### ISSUE THREAD

**Issue by rolfsimoes** [e-sensing/sits#1464]
**Describe the bug**
Torch-based training functions materialize full datasets into dense `matrix` and `array` objects before training, instead of processing data in batches.

This results in multiple full copies of the dataset being held in memory simultaneously, leading to excessive memory usage for large training datasets.

---

**To Reproduce**

```r
samples <- readRDS("big_samples.rds") # large dataset

mae_model <- sits_pre_train(
  samples = samples,
  encoder_method = sits_mae(
      encoder_model = sits_lighttae(),
      mask_ratio = 0.5,
      epochs = 1,
      verbose = TRUE
  )
)
```

During execution, memory usage increases significantly due to creation of many intermediate variables. These objects can reach hundreds of MB to GB depending on dataset size (see issue #1463 for more details).

---

**Additional context**

Root cause:

Full dataset preprocessing instead of batch-wise processing

This pattern appears in multiple models, including:

* `sits_mlp`
* `sits_tempcnn`
* `sits_resnet`
* `sits_tae`
* `sits_lighttae`
* `sits_mae`

However, this issue affects more `sits_mae()` that typically requires larger datasets.

---

**Expected behavior**

Data should be processed on demand (in batches)

---

**Proposed fix**

Refactor training pipeline for `sits_mae()` to use lazy datasets:

* split using indices only (`train_idx`, `val_idx`)
* move feature extraction and normalization into `torch::dataset`
* implement `.getbatch()` for batch processing
* convert to `torch_tensor` only at batch level
